File: src/main/java/camera/PropertiesConfiguration.java
package camera;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.util.Properties;

class PropertiesConfiguration implements Configuration
{
    private final Properties properties = new Properties();

    private final File file;

    PropertiesConfiguration(String fileName) {
        file = new File(fileName);
        load();
    }

    private void load() {
        if (!file.exists()) {
            return;
        }
        try (final InputStream inputStream = new FileInputStream(file)) {
            properties.load(inputStream);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private void save() {
        try (final OutputStream outputStream = new FileOutputStream(file)) {
            properties.store(outputStream, null);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static String uriKey(int number) {
        return "camera." + number + ".uri";
    }

    @Override
    public String getCameraURI(int number) {
        return properties.getProperty(uriKey(number));
    }

    @Override
    public void setCameraURI(int number, String uri) {
        properties.setProperty(uriKey(number), uri);
        save();
    }
}

File: src/main/java/camera/PropertiesConfiguration.kt
package camera

import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.util.Properties

internal class PropertiesConfiguration(fileName: String) : Configuration
{
    private val properties = Properties()

    private val file = File(fileName)

    init {
        load()
    }

    private fun load() {
        if (!file.exists()) {
            return
        }
        ignoreThrows {
            FileInputStream(file).use { inputStream -> properties.load(inputStream) }
        }
    }

    private fun save() {
        ignoreThrows {
            FileOutputStream(file).use { outputStream -> properties.store(outputStream, null) }
        }
    }

    private fun uriKey(number: Int) = "camera.$number.uri"

    override fun getCameraURI(number: Int): String {
        return properties.getProperty(uriKey(number))
    }

    override fun setCameraURI(number: Int, uri: String?) {
        properties.setProperty(uriKey(number), uri)
        save()
    }
}
